### DISCUSSION TITLE
Users, sharing, permissions and management.

### DISCUSSION THREAD

**Discussion by WGandy** [mediacms-io/mediacms#489] [General]
When I saw the Example Cases listed on the main page of the repository here I thought "I've finally found what I'm looking for!". Namely, an open-source video hosting/streaming platform aimed at institutional use rather than home entertainment use. The second very large requirement is transcoding of videos for various platforms and networks. MediaCMS looks great for all of this. 

But now that I've gotten mediaCMS installed I'm a bit confused. 

Pulling from example A from Example Cases, let's say that I'm a teacher and have 2 classes. I want the students of the accounting class to be able to view video A and the beekeeping class to be able to view video B and all students to be able to view video C (class standards and procedures). I was expecting some kind of provision where I could create a group of users for the accounting class and another group for the beekeeping class and then simply share videos with the groups as needed. But it doesn't seem that there is this kind of ability to restrict content other than public vs private vs unlisted. In this respect, it seems like mediaCMS is a self hosted Youtube rather than a private CMS. 

Is there more to this part of the program that I am perhaps misunderstanding?

**Comment by mgogoulos** [comment]
Hi @WGandy , what you are aiming to achieve is a valid scenario that could be implemented with some work on MediaCMS. The software has been designed to support a number of scenarios, but it cannot support *any* scenario out of the box! As you are describing, many cases are perfectly fitting within the public vs unlisted vs private workflow. For the rest of them, I am offering paid services to help organizations and companies to implement their exact scenarios. I have limited capacity for the time being but I would still be available to implement it or even provide some hints on how this can be done

Regards

**Comment by jp-rad** [comment]
With the following fork, categories can be used to classify classes:
[https://github.com/jp-rad/mediacms-protected.git](https://github.com/jp-rad/mediacms-protected.git)

```
mkdir /home/mediacms.io && cd /home/mediacms.io/
git clone https://github.com/jp-rad/mediacms-protected.git mediacms
cd /home/mediacms.io/mediacms/ && bash ./install.sh
```

This fork adds **'members'** field to Category and adds **'protected'** to the state of Media.

The 'protected' media is similar the 'private' media.
But users associated with members of a Category can access the 'protected' media of that Category.

**For example.**
1) Add an "Accounting Classroom" category and a "Beekeeping Classroom" category.
2) Assigne a teacher and the accounting class students to the **members** of  the "Accounting Classroom" category.
3) Assigne a teacher and the beekeeping class students to the **members** of the "Beekeeping Classroom" category.
4) Assign the **category** of VideoA to "Accounting Classroom" and set the **state** to "protected".
5) Assign the **category** of VideoB to "Beekeeping Classroom" and set the **state** to "protected".
6) Set the **state** of VideoC to "public".
